File: CS/CustomDocumentOperationService/Services/ReportProvider.cs
using DevExpress.XtraReports.Services;
using DevExpress.XtraReports.UI;

namespace DocumentOperationServiceSample.Services {
    public class ReportProvider : IReportProvider {
        XtraReport IReportProvider.GetReport(string id, ReportProviderContext context) {

            return (id == "CategoriesReport") ? new CategoriesReport() : new XtraReport() ;
        }
    }
}

File: VB/CustomDocumentOperationService/Services/ReportProvider.vb
Imports DevExpress.XtraReports.Services
Imports DevExpress.XtraReports.UI

Namespace DocumentOperationServiceSample.Services
	Public Class ReportProvider
		Implements IReportProvider

		Private Function IReportProvider_GetReport(ByVal id As String, ByVal context As ReportProviderContext) As XtraReport Implements IReportProvider.GetReport

			Return If(id = "CategoriesReport", New CategoriesReport(), New XtraReport())
		End Function
	End Class
End Namespace
